CONTRIBUTING.md

```diff
@@ -423,7 +423,74 @@ If it's correct, copy it (and only it) to `pygmt/tests/baseline`.
+
+#### Using data version control ([dvc](https://dvc.org)) to manage test images
+
+As the baseline images are quite large blob files that can change often (e.g.
+with new GMT versions), it is not ideal to store them in `git` (which is meant
+for tracking plain text files). Instead, we will use [`dvc`](https://dvc.org)
+which is like `git` but for data. What `dvc` does is to store the hash (md5sum)
+of a file. For example, given an image file like `test_logo.png`, `dvc` will
+generate a `test_logo.png.dvc` plain text file containing the hash of the
+image. This `test_logo.png.dvc` file can be stored as usual on GitHub, while
+the `test_logo.png` file can be stored separately on our `dvc` remote at
+https://dagshub.com/GenericMappingTools/pygmt.
+
+To **pull** or sync files from the `dvc` remote to your local repository, use
+the commands below. Note how `dvc` commands are very similar to `git`.
+
+    dvc status  # should report any files 'not_in_cache'
+    dvc pull  # pull down files from DVC remote cache (fetch + checkout)
+
+Once the sync/download is complete, you should notice two things. There will be
+images stored in the `pygmt/tests/baseline` folder (e.g. `test_logo.png`) and
+these images are technically reflinks/symlinks/copies of the files under the
+`.dvc/cache` folder. You can now run the image comparison test suite as per
+usual.
+
+    pytest pygmt/tests/test_logo.py  # run only one test
+    make test  # run the entire test suite
+
+To **push** or sync changes from your local repository up to the `dvc` remote
+at DAGsHub, you will first need to set up authentication using the commands
+below. This only needs to be done once, i.e. the first time you contribute a
+test image to the PyGMT project.
+
+    dvc remote modify upstream --local auth basic
+    dvc remote modify upstream --local user "$DAGSHUB_USER"
+    dvc remote modify upstream --local password "$DAGSHUB_PASS"
+
+The configuration will be stored inside your `.dvc/config.local` file. Note
+that the $DAGSHUB_PASS token can be generated at
+https://dagshub.com/user/settings/tokens after creating a DAGsHub account
+(can be linked to your GitHub account). Once you have an account set up, please
+ask one of the PyGMT maintainers to add you as a collaborator at
+https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration before
+proceeding with the next steps.
+
+The entire workflow for generating or modifying baseline test images can be
+summarized as follows:
+
+    # Sync with both git and dvc remotes
+    git pull
+    dvc pull
+
+    # Generate new baseline images
+    pytest --mpl-generate-path=baseline pygmt/tests/test_logo.py
+    mv baseline/*.png pygmt/tests/baseline/
+
+    # Generate hash for baseline image and stage the *.dvc file in git
+    dvc add pygmt/tests/baseline/test_logo.png
+    git rm -r --cached 'pygmt/tests/baseline/test_logo.png'  # optional
+    git add pygmt/tests/baseline/test_logo.png.dvc
```

Comment:
When I test this branch by adding a new baseline image, I got this error. I believe it won't be a issue after we migrate all baseline images to DAGsHub.
```
dvc add pygmt/tests/baseline/test_basemap.png
Adding...
ERROR:  output 'pygmt/tests/baseline/test_basemap.png' is already tracked by SCM (e.g. Git).
    You can remove it from Git, then add to DVC.
        To stop tracking from Git:
            git rm -r --cached 'pygmt/tests/baseline/test_basemap.png'
            git commit -m "stop tracking pygmt/tests/baseline/test_basemap.png"
```

Reply:
Yep, that's why I added the optional `git rm -r --cached 'pygmt/tests/baseline/test_logo.png'` line in the CONTRIBUTING.md docs. We can slowly do the migration of images to DAGsHub later, a bit like so:

1. After this PR #1036, change recommended way of testing from `@check_figures_equal()` to `@pytest.mark.mpl_image_compare`
2. Bump minimum GMT version to 6.2.0
3. Fix all the test images that have changed, storing the new test images with dvc on DAGsHub
4. Optional - Migrate `@check_figures_equal` tests to `@pytest.mark.mpl_image_compare` (can prioritize the slow tests as reported in #835)
5. Optional - Fully deprecate `@check_figures_equal()`, removing it from codebase and documentation in CONTRIBUTING.md, also close https://github.com/matplotlib/pytest-mpl/pull/95?
6. Write new tests for new functionality using `@pytest.mark.mpl_image_compare` only

I think we definitely need a bit more practice and train our contributors on this new `dvc` way of making test plots. If we like this new style (give it a few weeks/months), it might be worth considering telling upstream GMT on how to store all the postscript files (bit of a shame though with https://github.com/GenericMappingTools/gmt/pull/3344) :slightly_smiling_face:

Reply:
7. Update the install instructions, because `pygmt.test()` won't work for users.
8. Maybe add the baseline images as a release asset when making releases.

Reply:
```
    dvc add pygmt/tests/baseline/test_logo.png
    git rm -r --cached 'pygmt/tests/baseline/test_logo.png'
```
As mentioned above, when I run `dvc add`, it gives me an error, then I need to run `git rm -r --cache` to remove the image from the git repo. After that, I need to run `dvc add` again to make it work.

I'm not sure how long the migration will take. If we expect it will take weeks or months, we need to improve the instructions.

When we migrate the old images to DAGsHub, we need to run:
```
    dvc add pygmt/tests/baseline/test_logo.png
    git rm -r --cached 'pygmt/tests/baseline/test_logo.png' 
    dvc add pygmt/tests/baseline/test_logo.png
    git add pygmt/tests/baseline/test_logo.png.dvc
```
when we add new baseline images, we need to run:
```
    dvc add pygmt/tests/baseline/test_logo.png
    git add pygmt/tests/baseline/test_logo.png.dvc
```

Reply:
How about swapping the order around so that `git rm -r --cached` is first?

```suggestion
    # Generate hash for baseline image and stage the *.dvc file in git
    git rm -r --cached 'pygmt/tests/baseline/test_logo.png'  # optional
    dvc add pygmt/tests/baseline/test_logo.png
    git add pygmt/tests/baseline/test_logo.png.dvc
```

Reply:
The `git rm` command will give an error if someone wants to add a new image (i.e., the image is not in the git repository). Perhaps change the comment `# optional` to `# only run this if migrating an existing image from git to dvc`?

Reply:
Ok, I'll change the comment to read that (but shorten it a bit).